docs/concepts/11-versions.md

```diff
@@ -0,0 +1,609 @@
+
+4. **Version Progression**: Ensure each version is greater than the previous
+   - ✅ `1.0.0` → `1.0.1` → `1.1.0` → `2.0.0`
+   - ❌ Don't skip versions unnecessarily
+
+### For Platform Users
+
+1. **Production Environments**: Use exact versions for stability
+   ```
+   Set Version: "1.2.3"  # Pinned version for production
+   ```
+
+2. **Development/Staging**: Use "Latest" automation for newest features
+   ```
+   Set Version: "Latest"  # Auto-deploy to stable releases
+   ```
+
+3. **Release Channels**: Use tilde patterns for controlled automatic deployments
+   ```
+   Set Version: "~1"    # Auto-deploy within major version
+   Set Version: "~1.2"  # Auto-deploy within minor version
+   ```
+
+4. **Infrastructure Testing**: Enable development releases on a channel
+   ```
+   Set Version: "~1.0"  # Then set Release Strategy to "Development Release"
+   ```
+   - Ideal for staging/dev environments that mirror production
+   - Automatically deploys and tests development releases against real cloud resources
+   - Provides early feedback on Terraform plans and compliance issues
+
+## Version Lifecycle
+
+```mermaid
+graph TD
+    A[Set version in massdriver.yaml] --> B{Publish Type}
+    B -->|mass bundle publish -d| C[Development Release<br/>1.2.0-dev.timestamp]
+    B -->|mass bundle publish| D[Stable Release<br/>1.2.0]
+    C --> E[Test in Staging]
+    E --> F{Ready for Release?}
+    F -->|No| G[Fix Issues & Republish Dev Release]
+    G --> C
+    F -->|Yes| D
+    D --> H[Available in UI]
+    H --> I[Package Deployment]
+    I --> J{Version Selection}
+    J -->|Exact| K[Pin to 1.2.0]
+    J -->|Channel| L[Use ~1.2]
+    J -->|Strategy| M[Stable/Development Release]
+```
+
+**Workflow Steps:**
+
+1. **Version Definition**: Set semantic version in `massdriver.yaml`
+2. **Development Release**: Optional development release publishing for testing (`-d` flag)
+3. **Testing Phase**: Validate development releases in staging environments
+4. **Stable Release**: Publish final immutable version
+5. **Package Selection**: Choose version strategy in UI dropdown
+6. **Deployment**: Package runs with selected version/automation rule
+
+## Version Fields and Properties
+
+Each package in Massdriver has several version-related fields that control how it behaves:
+
+### Package Version Fields
+
+- **`version`** - The version constraint (e.g., `"1.2.3"`, `"~1.2"`, `"latest"`)
+- **`currentVersion`** - The resolved semantic version that will be deployed next
+- **`deployedVersion`** - The version that was last successfully deployed to infrastructure
+- **`availableUpgrade`** - The newest version available for upgrade based on the constraint
+- **`releaseStrategy`** - Whether the package receives stable releases only or also development releases
+
+### Version Constraint Types
+
+**Exact Versions:**
+- `"1.2.3"` - Use exactly this version
+- `"2.0.0"` - Pin to specific release
+
+**Release Channels:**
```

Comment:
I think this is the 3rd time release channel functionality is called out in this document. It should just be consolidated and referenced instead of repeated.